Matt,

I updated the fixver for CASSANDRA-15868 to include 3.11.  Let's handle everything there.

Works for me, thanks!

Reopening this one, as doing a major Netty upgrade in a minor 3.11.x is a more questionable proposition than doing a minor Netty upgrade in a major 4.0 C*.

I understand the hesitancy. What's the path forward here then? If we can determine what the bar would be to accept this change, I'd be glad to do what is necessary to meet it.

My suggestion was to run unit tests and dtests and add the results here. (Running cassandra-stress I've so far seen no issues, we just need to add proof.)

[https://github.com/apache/cassandra/pull/634]

[~mattsplat] running tests, dtests, and perhaps java driver test suite against this would make me comfortable enough, personally. Though I don't have time right now to do it myself.

Added the unit test results. I couldn't see any difference between cassandra-3.11 with or without the updated dependency. 

This failed in both cases:
{code}
[junit-timeout] Testcase: testIndexMemtableSwitching(org.apache.cassandra.index.sasi.SASIIndexTest):	FAILED
[junit-timeout] expected:<0> but was:<1>
[junit-timeout] junit.framework.AssertionFailedError: expected:<0> but was:<1>
[junit-timeout] 	at org.apache.cassandra.index.sasi.SASIIndexTest.testIndexMemtableSwitching(SASIIndexTest.java:2379)
{code}
but I understand this test is known to be flaky, per Slack conversations.

I ran CI here: [https://ci-cassandra.apache.org/job/Cassandra-devbranch/206/]

and here: [https://app.circleci.com/pipelines/github/driftx/cassandra/51/workflows/4680a5da-8e32-474e-b01c-059728a30b6f]

 

I believe the teardown failures like [https://app.circleci.com/pipelines/github/driftx/cassandra/51/workflows/4680a5da-8e32-474e-b01c-059728a30b6f/jobs/595] are new.

Thanks [~brandon.williams] for running this. I don't have bandwidth right now to investigate the failures, so I'll leave this open for others to look into.

Bumping this, would be great to fix this security hole and pickup some free performance for everyone. I was able to get up to 27% more throughput with the more recent version. Thanks!

[~brandon.williams] - I may not understand the nature of the failures you saw in CI, it appears to me they are random and happen on a small number of test iterations, rather than specific tests failing. Is that correct? If so, makes it very difficult to track down why the change would cause failures.

Could you point me to a successful run for comparison?

I ran them again here: [https://ci-cassandra.apache.org/job/Cassandra-devbranch/230/]

A lot of those can probably be ignore, but I'm talking about the ones like this: [https://ci-cassandra.apache.org/job/Cassandra-devbranch/230/testReport/junit/dtest.materialized_views_test/TestMaterializedViews/test_view_metadata_cleanup/]
{code:java}
Unexpected error found in node logs (see stdout for full details). Errors: [WARN  [epollEventLoopGroup-2-8] 2020-08-05 17:37:18,666 AbstractChannelHandlerContext.java:311 - An exception 'java.lang.NullPointerException' [enable DEBUG level for full stacktrace] was thrown by a user handler's exceptionCaught() method while handling the following exception:
java.nio.channels.ClosedChannelException: null
{code}

This one: https://ci-cassandra.apache.org/job/Cassandra-devbranch/230/testReport/junit/dtest.native_transport_ssl_test/TestNativeTransportSSL/test_connect_to_ssl/

also looks a bit fishy.

Here's a random run from trunk to compare (also not completely successful, that's still rare): https://ci-cassandra.apache.org/job/Cassandra-trunk/258/

Thanks Brandon - if that's the case it may not be possible to integrate this without additional changes, and I wonder how much interest there is in making further changes for this.

One last check - I see the comparison is from trunk on 4.0, would the 3.11.x branch yield the same results?

Well, that's kind of moot in a couple of ways.  If we're going to break tests on trunk by merging 3.11, we're going to revert that.  Also it was decided in CASSANDRA-15868 not to include this in 3.11.  That said, if it can help troubleshoot, I saw your 3.11 PR there and made a branch to run CI on, which I did here:
https://ci-cassandra.apache.org/job/Cassandra-devbranch/231/

There's a teardown failure at https://ci-cassandra.apache.org/job/Cassandra-devbranch/231/testReport/junit/dtest.cqlsh_tests.test_cqlsh_copy/TestCqlshCopy/test_reading_max_insert_errors/ and the SSL failure also persists.

